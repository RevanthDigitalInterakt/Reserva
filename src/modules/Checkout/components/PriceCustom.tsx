import * as React from 'react';
import {
  Typography,
  Box,
  theme,
} from '@usereservaapp/reserva-ui';

export interface IpriceCustom {
  num: number;
  fontFamily: keyof typeof theme.fonts;
  sizeInterger: number;
  sizeDecimal: number;
  negative?: boolean;
  color?: keyof typeof theme.colors;
}
export const PriceCustom: React.FC<IpriceCustom> = ({
  num, fontFamily, sizeInterger, sizeDecimal, negative, color = 'preto',
}) => {
  const integerPart = (num: number) => Math.floor(num);
  const decimalPart = (num: number) => (`${num.toFixed(2)}`).split('.')[1];
  return (
    <Box flexDirection="row">
      <Box>
        {negative
          ? (
            <Typography color={color} fontFamily={fontFamily} fontSize={sizeInterger}>
              - R$
              {integerPart(num)}
              ,
            </Typography>
          )
          : (
            <Typography color={color} fontFamily={fontFamily} fontSize={sizeInterger}>
              R$
              {integerPart(num)}
              ,
            </Typography>
          )}

      </Box>
      <Box alignSelf="flex-start">
        <Typography color={color} fontFamily={fontFamily} fontSize={sizeDecimal}>{decimalPart(num)}</Typography>
      </Box>
    </Box>
  );
};
