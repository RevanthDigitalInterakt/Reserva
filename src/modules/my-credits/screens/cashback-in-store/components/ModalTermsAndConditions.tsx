import LottieView from 'lottie-react-native';
import React from 'react';
import { ScrollView } from 'react-native';
import Modal from 'react-native-modal';
import { Box, Button, Icon, Typography } from 'reserva-ui';
import { loadingSpinner } from 'reserva-ui/src/assets/animations';


interface IModal {
  isVisible: boolean;
  loading: boolean;
  setIsVisible: () => void;
  setTermAndConditions: () => void;
  isAccepted: boolean;
}

export const ModalTermsAndConditions = ({
  isVisible,
  setIsVisible,
  setTermAndConditions,
  isAccepted,
  loading,
}: IModal) => (
  <Modal avoidKeyboard onBackdropPress={setIsVisible} isVisible={isVisible}>
    <Box bg="white" marginY="xxl" justifyContent="center" px="xxxs" py="xxxs">
      <Box position="absolute" top={16} right={20} zIndex={4}>
        <Button
          onPress={setIsVisible}
          variant="icone"
          icon={<Icon size={12} name="Close" />}
        />
      </Box>
      <Box mt="xxxs">
        <Typography fontFamily="reservaSerifRegular" fontSize={20}>
          Termos e condições
        </Typography>
      </Box>
      <ScrollView>
        <Box mt="xxxs">
          <Typography fontFamily="nunitoRegular" fontSize={15}>
            1) O Cartão-Presente pode ser adquirido tanto nas lojas (Reserva e Reserva Mini) quanto on-line (usereserva.com);
          </Typography>
        </Box>
        <Box mt="xxxs">
          <Typography fontFamily="nunitoRegular" fontSize={15}>
            2) O Cartão-Presente poderá ser utilizado em lojas da marca Reserva (exceto franquias) e Reserva Mini (incluindo on-line). Nas lojas, o valor mínimo para aquisição do Cartão-Presente é de R$ 100,00 (cem reais) e o máximo de R$ 5.000,00 (cinco mil reais). On-line, o Cartão pode ser adquirido com saldo de R$ 150,00 (cento e cinquenta reais), R$ 200,00 (duzentos reais), R$ 300,00 (trezentos reais), ou qualquer outro valor entre R$ 400,00 (quatrocentos reais) e R$ 5.000,00 (cinco mil reais);
          </Typography>
        </Box>
        <Box mt="xxxs">
          <Typography fontFamily="nunitoRegular" fontSize={15}>
          3) É imprescindível a apresentação do Cartão-Presente nas lojas, no momento da utilização;
          </Typography>
        </Box>
        <Box mt="xxxs">
          <Typography fontFamily="nunitoRegular" fontSize={15}>
          4) Em caso de aquisição on-line, deverá ser apresentado/digitado o código e senha do Cartão-Presente para sua utilização;
          </Typography>
        </Box>
        <Box mt="xxxs">
          <Typography fontFamily="nunitoRegular" fontSize={15}>
          5) No ato da compra do Cartão-Presente, o cliente receberá um cupom não fiscal;
          </Typography>
        </Box>
        <Box mt="xxxs">
          <Typography fontFamily="nunitoRegular" fontSize={15}>
          6) As mercadorias retiradas com o Cartão-Presente estarão sujeitas à política de trocas vigente nas lojas;

          </Typography>
        </Box>
        <Box mt="xxxs">
          <Typography fontFamily="nunitoRegular" fontSize={15}>
          7) Não há devolução de troco nas compras realizadas com o Cartão-Presente, ficando o saldo remanescente disponível no próprio cartão, dentro do prazo previsto no item “8”;

          </Typography>
        </Box>
        <Box mt="xxxs">
          <Typography fontFamily="nunitoRegular" fontSize={15}>
          8) O Cartão-Presente não é recarregável e tem prazo de 12 (doze) meses, contados a partir do dia de sua compra. Após este prazo, o cartão não poderá ser utilizado, expirando o saldo eventualmente existente;

          </Typography>
        </Box>
        <Box mt="xxxs">
          <Typography fontFamily="nunitoRegular" fontSize={15}>
          9) É de inteira responsabilidade do cliente a divulgação do código e senha do Cartão-Presente, bem como outras informações/materiais recebidos;

          </Typography>
        </Box>
        <Box mt="xxxs">
          <Typography fontFamily="nunitoRegular" fontSize={15}>
          10) O Cartão-Presente não é uma garantia bancária e não pode ser convertido em dinheiro ou qualquer outro meio de pagamento, tampouco ter o seu saldo transferido para outro cartão (nem mesmo em caso de perda/furto/roubo do cartão);

          </Typography>
        </Box>
        <Box mt="xxxs">
          <Typography fontFamily="nunitoRegular" fontSize={15}>
          11) O cliente poderá consultar o saldo do Cartão-Presente nas lojas Reserva, Reserva Mini ou através do site usereserva.com/cartaopresente;

          </Typography>
        </Box>
        <Box mt="xxxs">
          <Typography fontFamily="nunitoRegular" fontSize={15}>
          12) O cliente deverá recusar/devolver o Cartão-Presente no momento da sua compra/recebimento, em caso de constatação de violação ou dano;

          </Typography>
        </Box>
        <Box mt="xxxs">
          <Typography fontFamily="nunitoRegular" fontSize={15}>
          13) O possuidor do Cartão-Presente declara que está de acordo com os “Termos e Condições de Uso do Cartão-Presente”. Estes podem ser encontrados em todas as lojas e no site usereserva.com.

          </Typography>
        </Box>
        <Box mt="xxxs">
          <Typography fontFamily="nunitoRegular" fontSize={15}>
          14) Em caso de dúvidas ou consulta sobre prazo de validade do Cartão-Presente, entre em contato conosco através dos telefones (21) 2108-4990 (RJ) ou (11) 2388-8280 (outros estados). Se preferir, envie um e-mail para sac@usereserva.com.
          </Typography>
        </Box>
      </ScrollView>
      <Box width="100%" mt="micro">
        <Button
          bg="verdeSucesso"
          width="100%"
          height={50}
          disabled={isAccepted}
          onPress={setTermAndConditions}
        >
          {loading ? (
            <LottieView
              source={loadingSpinner}
              style={{
                width: 30,
              }}
              autoPlay
              loop
            />
          ) : (
            <Typography color="white" fontFamily="nunitoSemiBold" fontSize={13}>
              {isAccepted ? 'ACEITO' : 'ACEITAR'}
            </Typography>
          )}
        </Button>
      </Box>
    </Box>
  </Modal>
);
