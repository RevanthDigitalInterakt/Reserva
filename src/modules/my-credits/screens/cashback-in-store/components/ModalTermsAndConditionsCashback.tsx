import React from 'react';
import { ScrollView } from 'react-native';
import Modal from 'react-native-modal';
import {
  Box, Button, Icon, Typography,
} from '@usereservaapp/reserva-ui';

interface IModal {
  isVisible: boolean;
  setIsVisible: () => void;
  setTermAndConditions: () => void;
}

export function ModalTermsAndConditionsCashback({
  isVisible,
  setIsVisible,
  setTermAndConditions,
}: IModal) {
  return (
    <Modal avoidKeyboard onBackdropPress={setIsVisible} isVisible={isVisible}>
      <Box bg="white" marginY="xxl" justifyContent="center" px="xxxs" py="xxxs">
        <Box position="absolute" top={16} right={20} zIndex={4}>
          <Button
            onPress={setIsVisible}
            variant="icone"
            icon={<Icon size={12} name="Close" />}
          />
        </Box>
        <Box>
          <Typography
            fontFamily="reservaSerifBold"
            textAlign="center"
            fontSize={14}
          >
            AR&CO
          </Typography>
        </Box>
        <Box>
          <Typography
            fontFamily="reservaSerifRegular"
            textAlign="center"
            fontSize={20}
          >
            REGULAMENTO PARA ACÚMULO DE CASHBACK
          </Typography>
        </Box>
        <Box>
          <Typography
            fontFamily="reservaSerifBold"
            textAlign="center"
            fontSize={14}
          >
            12.09.2022
          </Typography>
        </Box>
        <ScrollView>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoBold" fontSize={15}>
              1. O PROGRAMA
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              1.1 O Programa foi criado e administrado por AR&CO, trata-se de um
              sistema de recompensas que funciona na modalidade de acúmulo e
              resgate de créditos de Cashback, com o objetivo de recompensar e
              valorizar o relacionamento com os seus clientes conforme os termos,
              cláusulas e condições a seguir estipuladas.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              1.2 No Programa os membros possuem resgate de Cashback para produtos
              AR&CO (Reserva, Reserva Mini, Reserva Go e Reversa), adquiridos em
              loja física.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              1.3 Não há qualquer taxa de inscrição, ingresso ou permanência no
              Programa.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              1.4 O Programa está disponível para clientes de todo o território
              nacional (Brasil), maiores de 18 (dezoito) anos.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              1.5 Ao ingressar no Programa, o cliente declara ter conhecimento
              deste regulamento, e ao cadastrar- se, estará pontuando e aderindo
              às regras aqui descritas.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoBold" fontSize={15}>
              2. DEFINIÇÕES
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              2.1 Programa: Refere-se ao "Programa de Acúmulo de Cashback" que
              visa conceder aos clientes membros recompensas em Cashback, dentro
              da plataforma.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              2.2 Membro(s): Cliente participante do Programa.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              2.3 Regulamento: O presente regulamento que rege o Programa.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              2.4 Cashback: Significa dinheiro de volta. Quando o membro realizar
              uma compra em loja física, parte do investimento retorna para ele
              como créditos de Cashback. A quantidade é calculada sobre uma
              porcentagem em cima do valor final pago pela compra.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              2.5 APP RESERVA: Aplicativo e plataforma digital onde é possível
              aderir ao Programa, disponível e acessível gratuitamente.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoBold" fontSize={15}>
              3. COMO ADERIR AO PROGRAMA
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              3.1 Primeiro é necessário baixar o APP RESERVA, disponível para
              download nos sistemas operacionais Android e iOS.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              3.2 Criar uma conta única e intransferível com nome, sobrenome, e-
              mail, senha, gênero, número de CPF, data de nascimento, e número de
              telefone.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              3.3 Possuindo a conta no APP RESERVA, basta apresentar o QR Code de
              validação, gerado dentro do aplicativo, durante a finalização da
              compra no caixa da loja (Caso ainda não tenha informado CPF,
              telefone e e-mail, será solicitado no cadastro dentro do
              aplicativo).
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoBold" fontSize={15}>
              4. COMO FUNCIONA
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              4.1 Toda vez que o cliente finalizar uma compra, em qualquer loja
              própria e franquias presencialmente, irá acumular cashback caso
              possua o APP RESERVA baixado, sendo que terá de fazer a validação da
              compra por QR CODE gerado no momento em que a compra for finalizada.
            </Typography>
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              O valor de acúmulo de cashback será informado pelo atendente no
              caixa. E constará no APP RESERVA em até 7 (sete) dias.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoBold" fontSize={15}>
              I) Do acúmulo
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              a. O percentual de acúmulo pode variar conforme campanha, promoção,
              mês e etc., ele incide sobre o valor final das compras, em reais,
              excluindo produtos não inclusos no Programa e valores descontados
              antes do pagamento, em todos os pedidos realizados dentro do escopo
              de produtos das marcas elencadas no item 1.2 e das marcas de sellers
              (marcas externas que compõem o APP RESERVA, que podem ou não aderir
              ao programa de Cashback, o que será sinalizado no momento de
              consulta do produto).
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              b. Em caso de utilização de outros formatos de desconto vinculados a
              voucher/cupom e promoções, o acúmulo incidirá no valor final das
              compras.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              c. O acúmulo acontecerá exclusivamente dentro do APP RESERVA.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoBold" fontSize={15}>
              II) Do resgate
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              a. Na hora de fazer o pagamento, você poderá utilizar o valor
              disponível de forma rápida e fácil. Os créditos em Cashback poderão
              ser de até 33% (trinta e três por cento) do valor total do pedido,
              e com um resgate que engloba somente as marcas AR&CO (Reserva,
              Reserva Mini, Reserva Go e Reversa).
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              b. Os créditos do Cashback serão creditados na ABA CASHBACK do seu
              APP RESERVA, 7 (sete) dias após a compra.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              c. Em caso de trocas, pelo mesmo produto, apenas alterando tamanho,
              não haverá movimentação no Cashback.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              d. Em caso de devolução ou cancelamento da compra, o valor utilizado
              de Cashback será estornado ao membro. Este prazo está sujeito aos
              termos de uso do APP RESERVA.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoBold" fontSize={15}>
              III) Do créditos
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              a. Os créditos ficam consolidados na página “Ver minha carteira”, no
              tópico “Meu Cashback”, no seu Perfil do APP RESERVA, onde é possível
              consultar o extrato de acúmulos e resgates.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              b. Os créditos acumulados possuem validade de 30 (trinta) dias, caso
              não sejam utilizados dentro deste período, eles expiram e não podem
              ser resgatados.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoBold" fontSize={15}>
              5. DO USO INDEVIDO
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              5.1 O uso fraudulento do Programa ou em desacordo com este
              regulamento, determinará o cancelamento do usuário no Programa e a
              perda dos créditos acumulados, independentemente da adoção das
              medidas legais cabíveis para o ressarcimento de todos e quaisquer
              danos sofridos.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              5.2 A AR&CO reserva o direito de descontinuar o Programa eliminando
              todo crédito de seu(s) cliente(s) se o(s) mesmo(s) ou em parte
              for(em) adquirido(s) por meio(s) fraudulentos, erro na entrada de
              dados, roubo ou qualquer outro meio que não o oficial do Programa.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              5.2 A AR&CO reserva o direito de descontinuar o Programa eliminando
              todo crédito de seu(s) cliente(s) se o(s) mesmo(s) ou em parte
              for(em) adquirido(s) por meio(s) fraudulentos, erro na entrada de
              dados, roubo ou qualquer outro meio que não o oficial do Programa.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              5.3 A AR&CO reserva-se o direito de debitar, sem a anuência de
              respectivo titular, todos e quaisquer pontos do Programa creditados
              indevidamente (em razão de duplicidade, erro, dolo etc.).
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              5.4 É expressamente proibido qualquer tipo de comercialização dos
              créditos de Cashback, sendo que qualquer transmissão feita
              indevidamente será considerada fraudulenta.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoBold" fontSize={15}>
              6. VALIDADE DO PROGRAMA
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              6.1 O Programa terá validade por tempo indeterminado.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoBold" fontSize={15}>
              7. DA ALTERAÇÃO DO REGULAMENTO
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              7.1 A AR&CO reserva-se o direito de alterar o Programa a qualquer
              momento a seu exclusivo critério.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoBold" fontSize={15}>
              8. DO CANCELAMENTO DO PROGRAMA
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              8.1 O Programa poderá ser cancelado a qualquer momento a critério
              exclusivo da AR&CO. Neste caso, o cancelamento será informado nos
              canais públicos de comunicação como sites das marcas e plataforma
              digital RESERVA (https://www.usereserva.com/), com 30 (trinta) dias
              de antecedência, de forma que os membros do Programa tenham tempo
              hábil para uso de seus créditos de Cashback.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              8.2 Finalizado o período de 30 (trinta) dias mencionado no item
              anterior, os créditos não serão mais válidos. Nesta hipótese, dentre
              outras previstas no presente regulamento, a AR&CO estará isenta de
              quaisquer responsabilidades em decorrência da falta de uso de
              créditos pelos membros.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              8.3 O cancelamento do Programa, ainda que de forma imotivada, não
              gerará ao cliente qualquer tipo de indenização.
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoBold" fontSize={15}>
              9. PRIVACIDADE E NOTIFICAÇÕES POR E-MAIL
            </Typography>
          </Box>
          <Box mt="xxxs">
            <Typography fontFamily="nunitoRegular" fontSize={15}>
              9.1 A sua participação no Programa está sujeita à Política de
              Privacidade da AR&CO, e também no APP RESERVA. Antes de aderir no
              Programa, é recomendável que você analise a Política de Privacidade
              para que entenda como a AR&CO irá tratar os seus dados. Nesta
              Política listamos todos os direitos que você tem em relação aos seus
              dados e como pode exercê-los, inclusive caso queira deixar de
              receber nossas comunicações.
            </Typography>
          </Box>
        </ScrollView>
        <Box width="100%" mt="micro">
          <Button
            variant="primarioEstreitoOutline"
            width="100%"
            height={50}
            onPress={setTermAndConditions}
          >
            <Typography
              letterSpacing={1.6}
              color="preto"
              fontFamily="nunitoRegular"
              fontSize={13}
            >
              ACEITO
            </Typography>
          </Button>
        </Box>
      </Box>
    </Modal>
  );
}
