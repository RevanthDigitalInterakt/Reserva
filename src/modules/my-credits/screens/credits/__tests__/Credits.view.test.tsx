import React from 'react';
import TestRenderer from 'react-test-renderer';
import {
  Box,
  Button,
  Divider,
  Icon,
  theme,
  Typography,
} from '@danilomsou/reserva-ui';
import { ThemeProvider } from 'styled-components/native';

import { CreditsView, CreditsViewProps } from '../Credits.view';
import { PriceCustom } from '../../../../Checkout/components/PriceCustom';

const render = (props: CreditsViewProps) => {
  const testRenderer = TestRenderer.create(
    <ThemeProvider theme={theme}>
      <CreditsView {...props} />
    </ThemeProvider>
  );
  const testInstance = testRenderer.root;

  return {
    testInstance,
    testRenderer,
  };
};

describe('CreditsView', () => {
  describe('when creditsBalance is not null and screenCashback is true', () => {
    const props: CreditsViewProps = {
      creditsBalance: Math.floor(Math.random() * 100),
      screenCashbackInStoreActive: true,
      handleNavigateToCashbackInStore: jest.fn(),
    };
    const { testInstance } = render(props);

    it.skip('should render a PriceCustom', () => {
      const priceCustom = testInstance.findByType(PriceCustom);
      const typography = testInstance.findAllByType(Typography);

      expect(typography[0].props.children).toBe('Meus crÃ©ditos');
      expect(priceCustom.props.num).toBe(props.creditsBalance);
    });

    it.skip('should render a Button CashbackInStore and set onPress equal handleNavigate', () => {
      const typography = testInstance.findAllByType(Typography);
      const button = testInstance.findByType(Button);
      const icon = testInstance.findByType(Icon);

      expect(button.props.onPress).toBe(props.handleNavigateToCashbackInStore);
      expect(button.props.children).toBeTruthy();
      expect(typography[5].props.children).toBe('Cashback em Lojas');
      expect(icon.props.name).toBe('Cashback');
    });

    it.skip('should navigate to CashbackInStore when click on button', () => {
      const button = testInstance.findByType(Button);
      button.props.onPress();
      expect(props.handleNavigateToCashbackInStore).toHaveBeenCalled();
    });

    it.skip("should render a two Divider's", () => {
      const divider = testInstance.findAllByType(Divider);

      expect(divider).toHaveLength(2);
    });
  });
  describe('when all props required are passed', () => {
    const props: CreditsViewProps = {
      creditsBalance: 200,
      screenCashbackInStoreActive: true,
      handleNavigateToCashbackInStore: jest.fn(),
    };
    const { testRenderer } = render(props);

    it.skip('should match snapshot', () => {
      expect(testRenderer.toJSON()).toMatchSnapshot();
    });
  });
});
