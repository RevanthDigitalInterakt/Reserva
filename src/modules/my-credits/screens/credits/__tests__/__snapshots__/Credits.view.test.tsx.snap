// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditsView when all props required are passed should match snapshot 1`] = `
<View
  mt="sm"
  mx="xxs"
  style={
    Array [
      Object {
        "marginLeft": 24,
        "marginRight": 24,
        "marginTop": 40,
      },
    ]
  }
>
  <View
    mb="nano"
    style={
      Array [
        Object {
          "marginBottom": 8,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ReservaSerif-Regular",
            "fontSize": 28,
          },
        ]
      }
      variant="tituloSessoes"
    >
      Meus créditos
    </Text>
  </View>
  <Text
    fontFamily="nunitoRegular"
    fontSize={14}
    style={
      Array [
        Object {
          "fontFamily": "Nunito-Regular",
          "fontSize": 14,
        },
      ]
    }
  >
    Use o crédito na sua próxima compra. Ele aparecerá automaticamente no ato do pagamento.
  </Text>
  <View
    mt="xxs"
    style={
      Array [
        Object {
          "marginTop": 24,
        },
      ]
    }
  >
    <View
      backgroundColor="divider"
      height={1}
      style={
        Array [
          Object {
            "backgroundColor": "#D8D9DA",
            "height": 1,
          },
        ]
      }
      variant="fullWidth"
    />
    <View
      flexDirection="row"
      justifyContent="space-between"
      py="xxs"
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 24,
            "paddingTop": 24,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "ReservaSerif-Regular",
              "fontSize": 20,
            },
          ]
        }
        variant="subtituloSessoes"
      >
        Créditos e cashback
      </Text>
      <View
        flexDirection="row"
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            color="preto"
            fontFamily="nunitoBold"
            fontSize={20}
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 20,
                },
              ]
            }
          >
            R$ 
            200
            ,
          </Text>
        </View>
        <View
          alignSelf="flex-start"
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
              },
            ]
          }
        >
          <Text
            color="preto"
            fontFamily="nunitoBold"
            fontSize={13}
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 13,
                },
              ]
            }
          >
            00
          </Text>
        </View>
      </View>
    </View>
    <View
      backgroundColor="divider"
      height={1}
      style={
        Array [
          Object {
            "backgroundColor": "#D8D9DA",
            "height": 1,
          },
        ]
      }
      variant="fullWidth"
    />
    <View
      flexDirection="row"
      mt="xxs"
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginTop": 24,
          },
        ]
      }
    >
      <View
        alignItems="center"
        style={
          Array [
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <RNSVGSvgView
            bbHeight={20}
            bbWidth={20}
            color="preto"
            focusable={false}
            mr="xxxs"
            size={20}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Array [
                  Object {
                    "color": "#333333",
                    "height": 20,
                    "marginRight": 16,
                    "width": 20,
                  },
                ],
                Object {
                  "flex": 0,
                  "height": 20,
                  "width": 20,
                },
              ]
            }
          >
            <RNSVGGroup>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="100%"
                bbWidth="100%"
                focusable={false}
                height="100%"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {
                        "height": "100%",
                        "width": "100%",
                      },
                    ],
                    Object {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={18}
                vbWidth={18}
                width="100%"
              >
                <RNSVGGroup>
                  <RNSVGGroup
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        -20,
                        -683,
                      ]
                    }
                  >
                    <RNSVGGroup
                      fill={null}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          20,
                          683,
                        ]
                      }
                      propList={
                        Array [
                          "fill",
                          "stroke",
                        ]
                      }
                      stroke={4281545523}
                    >
                      <RNSVGCircle
                        cx="9"
                        cy="9"
                        propList={
                          Array [
                            "stroke",
                          ]
                        }
                        r="8"
                        stroke={4281545523}
                      />
                      <RNSVGCircle
                        cx="9"
                        cy="9"
                        fill={null}
                        propList={
                          Array [
                            "fill",
                          ]
                        }
                        r="8.5"
                      />
                    </RNSVGGroup>
                    <RNSVGPath
                      d="M171.52,270.232a.705.705,0,0,1,0-1.411,1.134,1.134,0,0,1,.763.369.705.705,0,1,0,.981-1.014,2.721,2.721,0,0,0-1.039-.65v-.822a.705.705,0,1,0-1.411,0v.827a2.116,2.116,0,0,0,.705,4.111.705.705,0,1,1,0,1.411,1.277,1.277,0,0,1-.916-.534.705.705,0,0,0-1.08.908,2.924,2.924,0,0,0,1.291.919v.823a.705.705,0,1,0,1.411,0v-.827a2.116,2.116,0,0,0-.705-4.111Zm0,0"
                      fill={4281545523}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          -142.496,
                          421.063,
                        ]
                      }
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
            </RNSVGGroup>
          </RNSVGSvgView>
          <Text
            fontFamily="nunitoBold"
            fontSize={16}
            style={
              Array [
                Object {
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 16,
                },
              ]
            }
          >
            Cashback em Lojas
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
